Meta/serenity.sh

```diff
@@ -221,9 +234,9 @@ build_target() {
     # With zero args, we are doing a standard "build"
     # With multiple args, we are doing an install/image/run
     if [ $# -eq 0 ]; then
-        cmake --build "$SUPER_BUILD_DIR"
+        cmake --build "$SUPER_BUILD_DIR" -- -j "$MAKEJOBS"
```

Comment:
This does not seem to have any effect, hardcoding in the superbuild cmakelists.txt works but seems awfully ugly (and non portable)

Reply:
@friendlyanon are you aware of any tasteful CMake tricks to pass `-j` or other job control flags down to external projects? Reading it from the environment and manually adding as the build step command or some such feels very heavy handed.

Reply:
There are a few options:

* Just use Ninja and that will use all cores by default
* Since CMake 3.16 is required by Serenity, you can just use the native `-j` flag added in 3.12 (`cmake --build ... -j ...`)

For getting the core count, I recommend using the [ProcessorCount](https://cmake.org/cmake/help/latest/module/ProcessorCount.html) module with a script:

```cmake
cmake_minimum_required(VERSION 3.1)
include(ProcessorCount)
ProcessorCount(N)
if(NOT N EQUAL "0") # avoid printing if couldn't get cores
  execute_process(COMMAND "${CMAKE_COMMAND}" -E echo "${N}")
endif()
```

```sh
cmake --build "$SUPER_BUILD_DIR" -j $(cmake -P .../cores.cmake) # unquoted, expands to nothing when no cores found
# or just set it as a variable and provide a fallback
cores="$(cmake -P .../cores.cmake)"
cmake --build "$SUPER_BUILD_DIR" -j ${cores:-${FALLBACK_CORES}}
```

I don't know the exact usecases, but these are what came to my mind right now, so use only what makes sense for your usecase.

Reply:
The intended use case would be for developers to limit the number of cores below $(nproc) or equivalent, so that they have CPU available to do other things with their system, like watch YouTube or game or w/e. So since we already use ninja by default the question is more how to pass down to the sub-projects from `cmake --build $SUPERBUILD_BINARY_DIR` that ninja shouldn't use all available cores when doing the build and install stages for the Lagom and serenity ExternalProjects

Reply:
Hm, right. There is no built-in way to do that. It's DIY time :)

```cmake
set(JOBS "")
if(DEFINED ENV{JOBS} AND "$ENV{JOBS}" MATCHES "^[0-9]+$" AND "$ENV{JOBS}" GREATER "0")
  set(JOBS "$ENV{JOBS}")
endif()

ExternalProject_Add(
    ...
    BUILD_COMMAND "${CMAKE_COMMAND}" --build <BINARY_DIR> --config "$<CONFIG>"
    -j ${JOBS} # unquoted
    ...
)
```

Something like that maybe?

Reply:
This works, but not quite how I envisioned it.
It doesn't update when building and takes the value at "generation" (not `--build` time) time for cmake only.
Can we pass a variable at build time or somehow force it to read the environment every time?

Reply:
Actually I found the `CMAKE_BUILD_PARALLEL_LEVEL` environment variable which is a fallback for `-j` so I just forward `MAKEJOBS` to that now.

Reply:
[CMAKE_BUILD_PARALLEL_LEVEL](https://cmake.org/cmake/help/latest/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html):

> _New in version 3.12._

Doh! Feels like a slap in the face, since I usually check the variables list first, but didn't do so this time.